src/critnib/critnib.c

```diff
@@ -290,8 +318,34 @@ static struct critnib_leaf *alloc_leaf(struct critnib *__restrict c) {
     }
 
     struct critnib_leaf *k = c->deleted_leaf;
-
     c->deleted_leaf = k->value;
+
+    // Verify if we are reusing a non-freed leaf.
+    // In such case (extremely unlikely) a memory leak might happen.
+    uint64_t ref_count;
+    utils_atomic_load_acquire_u64(&k->ref_count, &ref_count);
+    if (ref_count > 0) {
+        void *to_be_freed;
+        utils_atomic_load_acquire_ptr(&k->to_be_freed, &to_be_freed);
+        if (to_be_freed != NULL) {
+            void *key;
+            utils_atomic_load_acquire_ptr((void **)&k->key, &key);
+            LOG_DEBUG("WARNING: reusing probably non-freed leaf: key = %p, "
```

Comment:
nit: this message could be moved to #else so in debug there would be only one message

Reply:
This code has been removed

Reply:
Done